flake.nix

```diff
@@ -151,12 +149,11 @@
                     ++ allOutputs python3Packages.kernels
                     ++ lib.optionals stdenv.hostPlatform.isLinux (allOutputs stdenvGlibc_2_27)
                   );
-                buildSetLinkFarm =
-                  buildSet: pkgs.linkFarm (buildName buildSet.buildConfig) (buildSetOutputs buildSet);
+                buildSetLinkFarm = buildSet: pkgs.linkFarm buildSet.torch.variant (buildSetOutputs buildSet);
               in
               pkgs.linkFarm "packages-for-cache" (
                 map (buildSet: {
-                  name = buildName (buildSet.buildConfig);
+                  name = buildSet.torch.variant;
```

Comment:
I have moved generation of variant strings `torch-...` to the Torch derivation itself. This makes it easier to access it from everywhere.